lib/compat/experimental/blocks.php

```diff
@@ -63,6 +63,9 @@ function build_comment_query_vars_from_block( $block ) {
 				$comment_args['paged'] = ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
 			}
 		}
+		if ( empty($page) && $comment_args['paged'] ) {
+			set_query_var( 'cpage', $comment_args['paged'] );
+		}
```

Comment:
A few suggestions to make the code more readable and guarded for future.

- Move this conditional code into the `if` branch above. Why? It will only occur if `$per_page > 0`.
- Guard to ensure the `'paged'` key exists. Why? It's possible for `$comments_args['paged']` to not be set given the `elseif` logic and for any further changes. If that happens, an error will occur - severity dependent upon PHP version.
- As these variables represent integers, compare against `0`. Why? Makes the code more readable and understandable. How? It shows the intent of the code using the expected values instead of falsey and truthy checks.
- Add an inline comment to explain why this code is added. Without it, future contributors may remove or alter it without understanding why it exists.

Here's the suggested code within the `if` block:

```php
		if ( $per_page > 0 ) {
			$comment_args['number'] = $per_page;

			$page = (int) get_query_var( 'cpage' );
			if ( $page ) {
				$comment_args['paged'] = $page;
			} elseif ( 'oldest' === $default_page ) {
				$comment_args['paged'] = 1;
			} elseif ( 'newest' === $default_page ) {
				$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
			}

			// Set the `cpage` query var to ensure the previous and next pagination links are correct
			// when inheriting the Discussion Settings.
			if ( 0 === $page && isset( $comment_args['paged'] ) && $comment_args['paged'] > 0 ) {
				set_query_var( 'cpage', $comment_args['paged'] );
			}
		}

Reply:
Suggestions committed in https://github.com/WordPress/gutenberg/pull/39227/commits/de6b19d052fa2ebb1d560ca1df8dc17b2f35bd7c. Thanks @SantosGuillamot 👏